### PR TITLE
Pass setsockopt fd as int

### PR DESCRIPTION
**Pull request by tmds**
Even on 64-bit, Unix file descriptors are 32-bit.

CC @halter73

### CODE REVIEW

**Comment by halter73** [comment]
Thanks!

File: src/Kestrel.Transport.Sockets/SocketTransport.cs [hunk 1]
```diff
@@ -163,7 +163,7 @@ public Task StopAsync()
         }
 
         [DllImport("libc", SetLastError = true)]
-        private static extern int setsockopt(IntPtr socket, int level, int option_name, IntPtr option_value, uint option_len);
+        private static extern int setsockopt(int socket, int level, int option_name, IntPtr option_value, uint option_len);
```
Comment (halter73) [inline src/Kestrel.Transport.Sockets/SocketTransport.cs]: @tmds Do you think the same change should be made [in corefx](https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/Common/src/Interop/Unix/System.Native/Interop.SetSockOpt.cs#L17)?

Comment (tmds) [reply src/Kestrel.Transport.Sockets/SocketTransport.cs]: corefx converts the intptr to an int in native code

https://github.com/dotnet/corefx/blob/473463e644cba090257b6fde42a17af5bc01b01b/src/Native/Unix/Common/pal_utilities.h#L143-L148

Comment (tmds) [reply src/Kestrel.Transport.Sockets/SocketTransport.cs]: (corefx is using IntPtr between managed and native is because SafeHandles marshal to intptr)

**Pull request merged**